starfish/image/_stack.py

```diff
@@ -46,6 +46,10 @@ def __init__(self, image_partition):
                     raise TypeError("All tiles should have the same kind of dtype")
             if dtype.itemsize > max_size:
                 max_size = dtype.itemsize
+            if self._tile_shape is None:
+                self._tile_shape = tile.tile_shape
+            elif tile.tile_shape is not None and self._tile_shape != tile.tile_shape:
```

Comment:
Do you have to check if it is not none here since you already checked if it is none previously?

Reply:
Yup, we don't need the check.  Good call.

Reply:
Actually, I'm wrong.  First check is `self._tile_shape`, and the second check is `tile.tile_shape`

Reply:
oops missed that they were different, my fault